port_ocean/ocean.py

```diff
@@ -68,6 +71,26 @@ def __init__(
             integration_class(ocean) if integration_class else BaseIntegration(ocean)
         )
 
+    async def calculate_next_resync(self, now: datetime.datetime) -> float | None:
+        if self.config.runtime != "Saas" and not self.config.scheduled_resync_interval:
+            # There is no scheduled resync outside of Saas runtime or if not configured
+            return None
+
+        interval = self.config.scheduled_resync_interval
+        next_resync = None
+        if self.config.runtime == "Saas":
+            integration = await self.port_client.get_current_integration()
+            interval_str = (
+                integration.get("spec", {})
+                .get("appSpec", {})
+                .get("scheduledResyncInterval")
+            )
```

Comment:
I think we already have a code that turns the integration to an instance of integration, that way we don't need to actually use dictionary to get that key

Reply:
I think your referring to `ocean.config.integration` but it only contains data from the env variables config and integration identifier